### PR TITLE
TYP: PandasObject._cache

### PR DESCRIPTION
**Pull request by topper-123**
Move ``_cache: Dict[str, Any]`` fragment to PandasObject, where it belongs + related changes.

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -63,7 +65,7 @@ def __repr__(self) -> str:
         # Should be overwritten by base classes
         return object.__repr__(self)
 
-    def _reset_cache(self, key=None):
+    def _reset_cache(self, key: Optional[str] = None) -> None:
```
Comment (jbrockmendel) [inline pandas/core/base.py]: do we ever actually pass a key?

Comment (topper-123) [reply pandas/core/base.py]: Twice, both times in groupby/groupby.py.

**Comment by WillAyd** [comment]
Just out of curiosity why do you think this is better placed in PandasObject?

**Comment by topper-123** [comment]
My motivation is to have it set on the same class as _reset_cache is set on, because those two belong logically together. This would also that _cache attribute isnt set on a child class that is incompatible with _reset_cache.

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**